src/devices/SenseHat/SenseHatLedMatrix.cs

```diff
@@ -95,9 +144,300 @@ public static int PositionToIndex(int x, int y)
+        public void SetText(string text)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                StopTextAnimationTimer();
+                Fill(_textBackgroundColor);
+                lock (_lockTextRenderState)
+                {
+                    _textRenderState = null;
+                }
+
+                return;
+            }
+
+            if (_pixelFont == null)
+            {
+                _pixelFont = new SenseHatTextFont();
+            }
+
+            var renderMatrix = _pixelFont.RenderText(text);
+            SenseHatTextRenderState renderState;
+            lock (_lockTextRenderState)
+            {
+                // Create a new render state containing the render matrix for the new text
+                _textRenderState = new SenseHatTextRenderState(renderMatrix, _textColor, _textBackgroundColor, _textRotation);
+                renderState = _textRenderState;
+            }
+
+            RenderText(renderState);
+
+            StartOrStopTextScrolling();
+        }
+
+        /// <summary>
+        /// Text color when rendering text.
+        /// </summary>
+        [Property]
+        public Color TextColor
+        {
+            get
+            {
+                return _textColor;
+            }
+            set
+            {
+                _textColor = value;
+
+                // Apply new state. Lock because render state is nullable and assignment may not be atomic.
+                SenseHatTextRenderState? renderState;
+                lock (_lockTextRenderState)
+                {
+                    if (_textRenderState != null)
+                    {
+                        _textRenderState = _textRenderState.ApplyTextColor(value);
+                    }
+
+                    renderState = _textRenderState;
+                }
+
+                RenderText(renderState);
+            }
+        }
+
+        /// <summary>
+        /// Text background color when rendering text.
+        /// </summary>
+        [Property]
+        public Color TextBackgroundColor
+        {
+            get
+            {
+                return _textBackgroundColor;
+            }
+            set
+            {
+                _textBackgroundColor = value;
+
+                // Apply new state. Lock because render state is nullable and assignment may not be atomic.
+                SenseHatTextRenderState? renderState;
+                lock (_lockTextRenderState)
+                {
+                    if (_textRenderState != null)
+                    {
+                        _textRenderState = _textRenderState.ApplyTextBackgroundColor(value);
+                    }
+
+                    renderState = _textRenderState;
+                }
+
+                RenderText(renderState);
+            }
+        }
+
+        /// <summary>
+        /// Text scroll speed in pixels per second.
+        /// </summary>
+        [Property]
+        public double TextScrollPixelsPerSecond
```

Comment:
note: if you're isolating this class but would like to utilize PropertyAttribute (although I don't ask you to do it in this PR) with the class you'd need to create a wrapper for the TextScroller, i.e. SenseHatLedMatrixTextScroller which would be just a tiny class which allow you to only do text writing (drawing and text scroller shouldn't be used together anyway because it will just look like bogus pixels showing up)

Reply:
The design objective was to keep things simple for the user, and text simply starts scrolling as soon as the there is more than one character to display, using the default `TextScrollPixelsPerSecond` (1 pixel per second). Separating the scrolling out is a possibility of course. Something to consider for a possible generalization maybe, that it should be possible to retain "automatically scroll long text" for tiny LED displays, which could be a property set in a base class with whatever default makes sense for the LED display at hand.